seeker_test.go

```diff
@@ -26,15 +26,16 @@ func TestSeeker(t *testing.T) {
 	f.WriteString("Line 1\n")
 	f.WriteString("Line 2\n")
 
-	s := NewSeeker(f)
+	s := New(f)
 
 	if err = s.SeekToStart(); err != nil {
 		t.Fatal(err)
 	}
 
 	var str string
-	if err = s.ReadLine(func(buf *bytes.Buffer) {
+	if err = s.ReadLine(func(buf *bytes.Buffer) error {
```

Comment:
So the func passed to ReadLines (https://github.com/itsmontoya/seeker/blob/474907a3be1798e5ac976de8ee02e8e61c1942f0/seeker.go#L277) can end the iteration early

When an error is returned from the provided fn, that error will pass through ReadLine or ReadLines so the calling func is aware.